mhs/common/mhs_common/state/work_description.py

```diff
@@ -155,13 +173,33 @@ def __init__(self, persistence_store: pa.PersistenceAdaptor, store_data: dict):
         logger.info('016', 'Successfully updated work description to state store for {key}', {'key': self.message_key})
         return old_data
 
-    async def set_status(self, new_status: MessageStatus):
+    async def update(self):
```

Comment:
Are you missing test cases for this method?

Reply:
Updated to include the test cases for the logic in this method, then I also included one for the new `_deserialize_data` method as well